thirdparty/GeoModel/CMakeLists.txt

```diff
@@ -7,20 +7,21 @@
 # file, You can obtain one at http://mozilla.org/MPL/2.0/.
 
 # CMake include(s).
-include( FetchContent )
+include(FetchContent)
 
 # Tell the user what's happening.
-message( STATUS "Building GeoModel as part of the ACTS project" )
+message(STATUS "Building GeoModel as part of the ACTS project")
 
-set( GEOMODEL_VERSION "${_acts_geomodel_version}")
-
-set(GEOMODEL_SETUP_JSON OFF CACHE BOOL "Skip setting up json completely" )
-
-set(GEOMODEL_BUILD_GEOMODELG4 "${ACTS_BUILD_PLUGIN_GEANT4}" CACHE BOOL "Build the Geant4 plugin" )
+set(GEOMODEL_VERSION "${_acts_geomodel_version}")
+set(GEOMODEL_SETUP_JSON OFF CACHE BOOL "Skip setting up json completely")
+set(GEOMODEL_BUILD_GEOMODELG4 "${ACTS_BUILD_PLUGIN_GEANT4}" CACHE BOOL "Build the Geant4 plugin")
 
 # Declare where to get geomodel from.
 FetchContent_Declare(geomodel ${ACTS_GEOMODEL_SOURCE}
     PATCH_COMMAND git am ${CMAKE_CURRENT_SOURCE_DIR}/0001-Add-option-to-skip-setting-up-json-completely.patch)
 
+# Disable warnings flagged as errors
+set_property(DIRECTORY APPEND PROPERTY COMPILE_OPTIONS -Wno-error)
```

Comment:
I think moving https://github.com/acts-project/acts/blob/main/CMakeLists.txt#L175 will not really help because we set `-Werror` in the CI right now https://github.com/acts-project/acts/blob/0b9ed368c93ea752ed1751e7fe7095af6ce61090/.github/workflows/builds.yml#L100

In my experience coupling different CMake projects always results in these kinds of problems.

I also don't like turning flags off which could just be absent in the first place but I do not see a good way around this.
1. Keep it as it is, which should only modify the flags for the current directory as far as I understood?
2. Add another cmake options to wire in flags which are then only used for ACTS and not the dependencies. This again makes me feel like we are working around the issue of coupling different CMake projects.

What do you think @krasznaa ?

Reply:
While I'm not a huge fan of the directory scoping, the benefit is that it's uniformly applied and can't be forgotten (which it WILL be if it can be). Similarly, `-DCMAKE_CXX_FLAGS=-Werror` is there in the CI because I absolutely do not want even a single warning in `main`, from dependencies or otherwise.

OTOH, I agree we probably shouldn't apply our build flags to internal builds of the dependencies. In any case, we would need to work around possible issues in dependency headers with `SYSTEM` includes. 

I think this issue shows again that we should refrain from including third-party code in our build directly with `FetchContent`, unless we basically know it's compatible with our build (like `nlohmann_json` and `pybind11` are, for example).

So I personally see two solutions that are different from the ones you mentioned already:

1. Remove GeoModel from `thirdparty` and rely on `find_package` (my preference)
2. Patch GeoModel to not produce warnings with our compile flags.